manifests/server/config.pp

```diff
@@ -44,6 +44,15 @@
     }
   }
 
+  # Issue a warning if other options are used for disabling SSL: according to
+  # the template they will be ordered before other 'ssl-*' options and in turns
+  # leads MySQL to ignore the deactivation request.
+  if (!$options['mysqld']['ssl'] or $options['mysqld']['skip-ssl'] or $options['mysqld']['disable-ssl']) {
+    notify {'warn-ineffective-disable-options':
+      message => 'Use "ssl-disable" => true to disable SSL! Other options are not respected!'
```

Comment:
This is not necessary. The synonyms work according to the documentation.

https://dev.mysql.com/doc/refman/5.5/en/using-secure-connections.html
https://dev.mysql.com/doc/refman/5.6/en/using-secure-connections.html

> To prevent use of encryption and override other --ssl-xxx options, invoke the client program with --ssl=0 or a synonym (--skip-ssl, --disable-ssl)

https://dev.mysql.com/doc/refman/5.7/en/using-secure-connections.html

> To prevent use of encryption and override other --ssl-xxx options, invoke the client program with --ssl-mode=DISABLED, --ssl=0, or a synonym (--skip-ssl, --disable-ssl)

> To use an unencrypted connection, invoke the client with --ssl-mode=DISABLED, --ssl=0, or a synonym (--skip-ssl, --disable-ssl)